api/networkserver.proto

```diff
@@ -73,6 +73,14 @@ service NsEndDeviceRegistry {
     };
   };
 
+  // Reset resets device state.
+  rpc Reset(ResetEndDeviceRequest) returns (EndDevice) {
```

Comment:
It is not clear to me from the rpc name (`Reset()`) what the scope is.

Is it resetting the session? What is the state? MAC state?

Reply:
Maybe `ResetFactoryDefaults()` ?

Reply:
The idea is that this resets all NS-controlled mutable state, so it's close to recreating the device